### ISSUE TITLE
ApplovinMediationFacebookAdapter versions higher than 6.6.0.2 causes errors during archiving.

### ISSUE THREAD

**Issue by PaXLiCh** [AppLovin/AppLovin-MAX-Unity-Plugin#102]
Updated to latest AppLovin integration plugin
Unity 2020.3
Xcode 13.2.1 on latest macos.
ApplovinMediationFacebookAdapter updated to 6.9.0.1 or later to 6.9.0.2
and during archiving we getting errors:
```
ld: warning: Could not find or use auto-linked library 'swiftDispatch'
ld: warning: Could not find or use auto-linked library 'swiftObjectiveC'
ld: warning: Could not find or use auto-linked library 'swiftUIKit'
ld: warning: Could not find or use auto-linked library 'swiftDarwin'
ld: warning: Could not find or use auto-linked library 'swiftCompatibility50'
ld: warning: Could not find or use auto-linked library 'swiftSwiftOnoneSupport'
ld: warning: Could not find or use auto-linked library 'swift_Concurrency'
ld: warning: Could not find or use auto-linked library 'swiftCoreGraphics'
ld: warning: Could not find or use auto-linked library 'swiftFileProvider'
ld: warning: Could not find or use auto-linked library 'swiftCoreAudio'
ld: warning: Could not find or use auto-linked library 'swiftDataDetection'
ld: warning: Could not find or use auto-linked library 'swiftAVFoundation'
ld: warning: Could not find or use auto-linked library 'swiftCore'
ld: warning: Could not find or use auto-linked library 'swiftFoundation'
ld: warning: Could not find or use auto-linked library 'swiftCoreMedia'
ld: warning: Could not find or use auto-linked library 'swiftMetal'
ld: warning: Could not find or use auto-linked library 'swiftCoreImage'
ld: warning: Could not find or use auto-linked library 'swiftCompatibilityDynamicReplacements'
ld: warning: Could not find or use auto-linked library 'swiftCoreFoundation'
ld: warning: Could not find or use auto-linked library 'swiftCompatibility51'
ld: warning: Could not find or use auto-linked library 'swiftsimd'
ld: warning: Could not find or use auto-linked library 'swiftUniformTypeIdentifiers'
ld: warning: Could not find or use auto-linked library 'swiftCoreMIDI'
ld: warning: Could not find or use auto-linked library 'swiftQuartzCore'
Undefined symbols for architecture arm64:
  "__swift_FORCE_LOAD_$_swiftCompatibilityDynamicReplacements", referenced from:
      __swift_FORCE_LOAD_$_swiftCompatibilityDynamicReplacements_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCompatibilityDynamicReplacements_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCompatibility51", referenced from:
      __swift_FORCE_LOAD_$_swiftCompatibility51_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCompatibility51_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCompatibility50", referenced from:
      __swift_FORCE_LOAD_$_swiftCompatibility50_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCompatibility50_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftUniformTypeIdentifiers", referenced from:
      __swift_FORCE_LOAD_$_swiftUniformTypeIdentifiers_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftUniformTypeIdentifiers_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftAVFoundation", referenced from:
      __swift_FORCE_LOAD_$_swiftAVFoundation_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftAVFoundation_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftDataDetection", referenced from:
      __swift_FORCE_LOAD_$_swiftDataDetection_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftDataDetection_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCoreMIDI", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreMIDI_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreMIDI_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftsimd", referenced from:
      __swift_FORCE_LOAD_$_swiftsimd_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftsimd_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftFileProvider", referenced from:
      __swift_FORCE_LOAD_$_swiftFileProvider_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftFileProvider_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftMetal", referenced from:
      __swift_FORCE_LOAD_$_swiftMetal_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftMetal_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCoreImage", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreImage_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreImage_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCoreGraphics", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreGraphics_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreGraphics_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftQuartzCore", referenced from:
      __swift_FORCE_LOAD_$_swiftQuartzCore_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftQuartzCore_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftDispatch", referenced from:
      __swift_FORCE_LOAD_$_swiftDispatch_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftDispatch_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftCoreFoundation", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreFoundation_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreFoundation_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftFoundation", referenced from:
      __swift_FORCE_LOAD_$_swiftFoundation_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftFoundation_$_FBAudienceNetwork)
  "_swift_getInitializedObjCClass", referenced from:
      static FBAudienceNetwork.FBAdFont.sfuiTextRegular(withSize: CoreGraphics.CGFloat) -> __C.UIFont in FBAudienceNetwork(FBAudienceNetwork.o)
      static FBAudienceNetwork.FBAdFont.sfuiDisplaySemibold(withSize: CoreGraphics.CGFloat) -> __C.UIFont in FBAudienceNetwork(FBAudienceNetwork.o)
      type metadata accessor for FBAudienceNetwork.FBAdFont in FBAudienceNetwork(FBAudienceNetwork.o)
  "__swift_FORCE_LOAD_$_swiftObjectiveC", referenced from:
      __swift_FORCE_LOAD_$_swiftObjectiveC_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftObjectiveC_$_FBAudienceNetwork)
  "_swift_once", referenced from:
      FBAudienceNetwork.FBAdFont.fontSizeTiny.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.fontSizeExtraSmall.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.fontSizeSmall.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.fontSizeMedium.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.fontSizeLarge.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.fontSizeExtraLarge.unsafeMutableAddressor : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
  "__swift_FORCE_LOAD_$_swiftCoreAudio", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreAudio_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreAudio_$_FBAudienceNetwork)
  "__swift_FORCE_LOAD_$_swiftDarwin", referenced from:
      __swift_FORCE_LOAD_$_swiftDarwin_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftDarwin_$_FBAudienceNetwork)
  "_swift_getObjectType", referenced from:
      FBAudienceNetwork.FBAdFont.init() -> FBAudienceNetwork.FBAdFont in FBAudienceNetwork(FBAudienceNetwork.o)
      FBAudienceNetwork.FBAdFont.__deallocating_deinit in FBAudienceNetwork(FBAudienceNetwork.o)
  "__swift_FORCE_LOAD_$_swiftUIKit", referenced from:
      __swift_FORCE_LOAD_$_swiftUIKit_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftUIKit_$_FBAudienceNetwork)
  "value witness table for Builtin.UnknownObject", referenced from:
      full type metadata for FBAudienceNetwork.FBAdFont in FBAudienceNetwork(FBAudienceNetwork.o)
  "_swift_getObjCClassMetadata", referenced from:
      @objc static FBAudienceNetwork.FBAdFont.sfuiTextRegular(withSize: CoreGraphics.CGFloat) -> __C.UIFont in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.sfuiDisplaySemibold(withSize: CoreGraphics.CGFloat) -> __C.UIFont in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.fontSizeTiny.getter : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.fontSizeExtraSmall.getter : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.fontSizeSmall.getter : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.fontSizeMedium.getter : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      @objc static FBAudienceNetwork.FBAdFont.fontSizeLarge.getter : CoreGraphics.CGFloat in FBAudienceNetwork(FBAudienceNetwork.o)
      ...
  "__swift_FORCE_LOAD_$_swiftCoreMedia", referenced from:
      __swift_FORCE_LOAD_$_swiftCoreMedia_$_FBAudienceNetwork in FBAudienceNetwork(FBAudienceNetwork.o)
     (maybe you meant: __swift_FORCE_LOAD_$_swiftCoreMedia_$_FBAudienceNetwork)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
To fix this we just reverting in dependencies.xml for this adapter pod version to 6.6.0.2

**Comment by santoshbagadi** [comment]
Hi @PaXLiCh , the latest [Facebook Audience Network SDK (6.9.0+) requires Xcode 13](https://developers.facebook.com/docs/audience-network/setting-up/platform-setup/ios/changelog) to build apps. Please update to the latest Xcode to fix the issue (https://github.com/AppLovin/AppLovin-MAX-Unity-Plugin/issues/94).

**Comment by PaXLiCh** [comment]
we already using xcode 13.2.1 on latest macos

**Comment by santoshbagadi** [comment]
The latest FB SDK also depends on a bunch of Swift libraries and they are not being pulled in correctly. For FB, it is generally added by the FB Unity plugin which is recommended to be imported in our [docs](https://dash.applovin.com/documentation/mediation/unity/mediation-adapters?network=FACEBOOK_MEDIATE#facebook-data-processing-options).

But, we have a way to automatically add Swift support to a project. Could you try updating the  `SwiftLanguageNetworks` in `MaxPostProcessBuildiOS.cs` file as follows and see if that fixes the issue. 

        private static readonly List<string> SwiftLanguageNetworks = new List<string>
        {
            "Facebook",
            "MoPub"
        };

**Comment by PaXLiCh** [comment]
We already using facebook unity sdk 11. However, edited MaxPostProcessBuildiOS.cs at least solves the problem with the assembly, we are still checking the functionality with the new version of the adapter.

**Comment by PaXLiCh** [comment]
properly tested now. it works.

**Comment by thomasmso** [comment]
@PaXLiCh - We released it in an official release v5.0.0 fyi.
